python/cudf/cudf/utils/gpu_utils.py

```diff
@@ -24,7 +24,28 @@ def validate_setup(check_dask=True):
 
     try:
         gpus_count = getDeviceCount()
-    except CUDARuntimeError:
+    except CUDARuntimeError as e:
+        if e.status in (
+            3,
+            35,
+            98,
+            100,
+            101,
+            127,
+            200,
+            201,
+            210,
+            304,
+            800,
+            801,
+            802,
+            803,
+            804,
+        ):
```

Comment:
This should be a set instead of a tuple. Additionally, I'm wondering if it would make sense to define a `cpdef enum` for these instead so we can refer to them by name instead of just by these values.

Reply:
Agree should be by name. Also does it make more sense to do cuda error handling at the C++ level and just propagate the human readable part up?

Reply:
Changed the error status validation by a name and Pushed the error handling to cython layer